src/python/pants/option/config_test.py

```diff
@@ -22,72 +38,175 @@ def setUp(self) -> None:
       disclaimer:
         Let it be known
         that.
-      blank_section:
 
       [a]
       list: [1, 2, 3, %(answer)s]
-      listappend: +[7, 8, 9]
+      list2: +[7, 8, 9]
```

Comment:
I renamed this because the syntax I'll be proposing for TOML appends is:

```toml
list2.append = [7, 8, 9]
```

(TOML/JSON/YAML don't support `+[` natively, as expected. We could require users to use a string to represent this like `"+[7, 8, 9]", but then we lose the benefits of first-class TOML like editor syntax highlighting and validation.)

Reply:
I love the `.append` syntax and would love to see it backported to our ini parsing as well if possible!

Reply:
Yes I really like it too! Much more explicit!